karton/system/system.py

```diff
@@ -177,66 +180,96 @@ def route_task(self, task: Task) -> None:
                 self.backend.increment_metrics(
                     KartonMetrics.TASK_ASSIGNED, identity, pipe=pipe
                 )
-
-        # Directly update the task status to be finished
-        self.backend.set_task_status(task, TaskState.FINISHED, pipe=pipe)
         pipe.execute()
 
+    def handle_tasks(self, task_uids: List[str]) -> None:
+        tasks = self.backend.get_tasks(task_uids)
+        binds = self.backend.get_binds()
+        for task in tasks:
+            if task is None:
+                raise RuntimeError(
+                    "Task disappeared while popping, this should never happen"
+                )
+
+            self.route_task(task, binds)
+            task.last_update = time.time()
+            task.status = TaskState.FINISHED
+            # Directly update the unrouted task status to be finished
+        self.backend.register_tasks(tasks)
+
+    def handle_operations(self, bodies: List[str]) -> None:
+        """
+        Left for backwards compatibility with Karton <=4.3.0.
+        Earlier versions delegate task status change to karton.system.
+        """
+        operation_bodies = []
+        tasks = []
+        for body in bodies:
+            operation_body = json.loads(body)
+            task = Task.unserialize(operation_body["task"])
+            new_status = TaskState(operation_body["status"])
+            if task.status != new_status:
+                task.last_update = time.time()
+                task.status = new_status
+                tasks.append(task)
+            operation_bodies.append(operation_body)
+
+        self.backend.register_tasks(tasks)
+        self.backend.produce_logs(
+            operation_bodies, logger_name=KARTON_OPERATIONS_QUEUE, level="INFO"
+        )
+
+    def process_routing(self) -> None:
+        # Order does matter! task dispatching must be before
+        # karton.operations to avoid races. Timeout must be shorter
+        # than self.gc_interval, but not too long allowing graceful shutdown
+        data = self.backend.consume_queues(
+            [KARTON_TASKS_QUEUE, KARTON_OPERATIONS_QUEUE], timeout=5
+        )
+        if data:
+            queue, body = data
+            if not isinstance(body, str):
+                body = body.decode("utf-8")
+            if queue == KARTON_TASKS_QUEUE:
+                tasks = [body] + self.backend.consume_queues_batch(queue, 100)
```

Comment:
```suggestion
                tasks = [body] + self.backend.consume_queues_batch(queue, max_count=100)
```
In another note, the number seems a bit arbitrary